src/bit_encoding/bitwriter.rs

```diff
@@ -109,3 +109,42 @@ impl<W: io::Write> BitWriter<W> {
         Ok(len)
     }
 }
+
+// Implement for io::Sink to make vec() an associated function
+// Implementing for a generic W would require callers to specify W, which is annoying
```

Comment:
In fc91f4e32788f03d6d303fdc681704c68a3065a9:

This "implement on `BitWriter<io::Sink>` trick is pretty clever but I think it's too clever :). We should just make this a standalone function.